table.go

```diff
@@ -496,6 +496,9 @@ func (t *Table) RotateBlock(ctx context.Context, block *TableBlock, skipPersist
 	defer commit()
 
 	id := generateULID()
+	for id.Time() == block.ulid.Time() { // Ensure the new block has a different timestamp.
+		id = generateULID()
```

Comment:
I like to put a `runtime.Gosched` in these loops to yield the cpu. Up to you

Reply:
good call